vortex-array/src/array/primitive/mod.rs

```diff
@@ -151,10 +153,41 @@ impl PrimitiveArray {
         self.buffer().typed_data()
     }
 
+    pub fn patch(self, patch: &dyn Array) -> VortexResult<Self> {
+        match patch.encoding().id() {
```

Comment:
I wonder if it's worth making the caller do this downcast? Could even take `patch<P: PrimInt, T: NativePType>(self, positions: &[P], values: &[T])`

Reply:
Hmmmmm, ok I guess this makes it obvious you're patching with a sparse array and if it's not sparse you likely want a new function